### PR TITLE
circleci: extend coverage some more and disable travis

### PR DESCRIPTION
**Pull request by Habbie**
### Short description

* switch to clang-8
* test Recursor SNMP, DNSTAP, Protobuf
* enable asan+ubsan for all products
* fixes bugs that came out of the previous two items

Missing:
* ~ubsan for rec~
* ~asan+ubsan for auth (#9835)~
* ~`--enable-option-checking=fatal`~
* ~remotebackend: see if there is a more modern iteration form that does not break on asan~ moved to #9915 

Problem:
* ~recursor bulktest appears to be failing~
* ~`==15170==WARNING: invalid path to external symbolizer!` (works on my system, but not in CircleCI, need to figure out why)~

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
> recursor bulktest appears to be failing

Disabling asan makes it pass again.

**Comment by Habbie** [comment]
#9854 

#9480 (plus backport in #9853 )

**Comment by Habbie** [comment]
#9854 did not fix the bulktest crashes. Not investigated yet.

**Comment by rgacogne** [comment]
> #9854 did not fix the bulktest crashes. Not investigated yet.

For the record, it should be fixed by 07ddb3678657e674f7191c54ac647280a9d880c3.

**Comment by Habbie** [comment]
Rebased after #9843

File: modules/lmdbbackend/lmdbbackend.cc [hunk 1]
```diff
@@ -824,7 +824,10 @@ bool LMDBBackend::getDomainInfo(const DNSName &domain, DomainInfo &di, bool getS
       serFromString(val.get<string_view>(), rr);
 
       if(rr.content.size() >= 5 * sizeof(uint32_t)) {
-        uint32_t serial = *reinterpret_cast<uint32_t*>(&rr.content[rr.content.size() - (5 * sizeof(uint32_t))]);
+        uint32_t serial;
+        // a SOA has five 32 bit fields, the first of which is the serial
+        // there are two variable length names before the serial, so we calculate from the back
+        memcpy(&serial, reinterpret_cast<uint32_t*>(&rr.content[rr.content.size() - (5 * sizeof(uint32_t))]), sizeof(serial));
```
Comment (mind04) [inline modules/lmdbbackend/lmdbbackend.cc:830]: The reinterpret_cast<uint32_t*> is redundant.  The type is irrelevant for memcpy

**Comment by rgacogne** [comment]
The protocol buffer failure on Circle CI is due to the fact that the protozero code does not set the `inBytes` field for zero-length messages, while the existing Protocol Buffer code did, and the regression tests expect that field to be present.

**Comment by Habbie** [comment]
> The protocol buffer failure on Circle CI is due to the fact that the protozero code does not set the `inBytes` field for zero-length messages, while the existing Protocol Buffer code did, and the regression tests expect that field to be present.

Temporarily cherrypicked that fix to test it.

**Comment by rgacogne** [comment]
So just for kicks:
```
diff --git modules/geoipbackend/geoipbackend.cc modules/geoipbackend/geoipbackend.cc
index 5aaa06b47..090b91228 100644
--- modules/geoipbackend/geoipbackend.cc
+++ modules/geoipbackend/geoipbackend.cc
@@ -160,8 +160,17 @@ void GeoIPBackend::initialize() {
       DNSName qname = DNSName(recs->first.as<string>());
       vector<GeoIPDNSResourceRecord> rrs;
 
-      for(YAML::Node item :  recs->second) {
-        YAML::const_iterator rec = item.begin();
+      auto&& range = recs->second;
+#if 1
+      /* does not crash */
+      auto item = recs->second.begin();
+#else
+      /* crashes */
+      auto item = range.begin();
+#endif
+      auto end = recs->second.end();
+      for(; item != end; item++) {
+        YAML::const_iterator rec = item->begin();
         GeoIPDNSResourceRecord rr;
         rr.domain_id = dom.id;
         rr.ttl = dom.ttl;
```
I don't think I want to get near `yaml-cpp` anymore..

**Comment by Habbie** [comment]
I believe this is ready!

**Review by rgacogne — APPROVED** [review]
:heart:

**Review by mind04 — APPROVED** [review]

**Pull request merged**